server/auth/auth_oauth.py

```diff
@@ -0,0 +1,176 @@
+
+    def is_valid(self):
+        return True
+
+    def requires_client_login(self):
+        return True
+
+    def add_url_rules(self, app):
+        app.add_url_rule("/login", "login", self.login, methods=["GET"])
+        app.add_url_rule("/logout", "logout", self.logout, methods=["GET"])
+        app.add_url_rule("/oauth2/callback", "callback", self.callback, methods=["GET"])
+
+    def complete_setup(self, flask_app):
+        self.oauth = OAuth(flask_app)
+        if self.callback_base_url is None:
+            # assume this is for a local test, and the oauth provider has been configured
+            # with a callback that understands a localhost callback (e.g. A http://localhost:5005).
+            server_config = flask_app.app_config.server_config
+            self.callback_base_url = f"http://{server_config.app__host}:{server_config.app__port}"
+
+        self.client = self.oauth.register(
+            "oauth",
+            client_id=self.client_id,
+            client_secret=self.client_secret,
+            api_base_url=self.api_base_url,
+            access_token_url=f"{self.api_base_url}/oauth/token",
+            authorize_url=f"{self.api_base_url}/authorize",
+            client_kwargs={
+                "scope" : "openid profile email",
+            }
+        )
+
+    def is_authenticated(self):
+        try:
+            payload = self.get_jwt_payload()
+            return payload is not None
+        except AuthenticationError:
+            return False
+
+    def get_userid(self):
+        payload = self.get_jwt_payload()
+        if payload and payload.get("sub"):
+            return payload.get("sub")
+        return None
+
+    def get_username(self):
+        payload = self.get_jwt_payload()
+        if payload and payload.get("name"):
+            return payload.get("name")
+        return None
+
+    def login(self):
+        callbackurl = f'{self.callback_base_url}/oauth2/callback'
+        return_path = request.args.get("dataset", "")
+        return_to = f"{self.callback_base_url}/{return_path}"
+        # save the return path in the session cookie, accessed in the callback function
+        session["oauth_callback_redirect"] = return_to
+        return self.client.authorize_redirect(redirect_uri=callbackurl)
+
+    def logout(self):
+        if self.CXG_ID_TOKEN in session:
+            del session[self.CXG_ID_TOKEN]
+        return_path = request.args.get("dataset", "")
+        return_to = f"{self.callback_base_url}/{return_path}"
+        params = {'returnTo' : return_to, 'client_id' : self.client_id}
+        return redirect(self.client.api_base_url + '/v2/logout?' + urlencode(params))
+
+    def callback(self):
+        token = self.client.authorize_access_token()
+        id_token = token.get("id_token")
+        session[self.CXG_ID_TOKEN] = id_token
+        r = session.get("oauth_callback_redirect", "/")
```

Comment:
Avoid single letter variable names which aren't readable.

Reply:
Okay, renamed it.